### DISCUSSION TITLE
Rewrite an existing page to a new route based on cookie

### DISCUSSION THREAD

**Discussion by skantipu** [vercel/next.js#40188] [Help]
Next.js rewrites seems to work when `source` is a new route. If the `source` maps to one of the existing pages, is it possible to rewrite it to a new route based on a cookie check? I tried multiple times, but if the `source` is mapping to one of the existing pages, then the rewrite is not being honored and the existing page is rendered. 

My current page structure
```
pages
  study
    hello-word
    visitor
       hello-world
```

Need `/study/hello-world` signed-out visitor page (without cookie) redirecting to a new destination path (say `/study/visitor/hello-world`). Is that possible? Please help. Thanks.

**Comment by fitiskin** [comment]
Do you need a redirect or a rewrite? Anyways if I understand you correctly, then consider using [middleware](https://nextjs.org/docs/advanced-features/middleware). You can:

* [read cookies](https://nextjs.org/docs/advanced-features/middleware#using-cookies) from the `request`:
  ```js
  export function middleware(request) {
    const cookie = request.cookies.get('authorized')
  
    if (!cookie) {
      //
    }

    //
  }
  ```
* then call [`rewrite` or `redirect` on `NextResponse`](https://nextjs.org/docs/advanced-features/middleware#nextresponse) depends on cookies
* and scope the middleware to specific paths using [matcher](https://nextjs.org/docs/advanced-features/middleware#matcher):
  ```js
  export const config = {
    matcher: '/study/hello-world',
  }
  ```

Hope it helps!
